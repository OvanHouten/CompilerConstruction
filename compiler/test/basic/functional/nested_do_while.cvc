extern void printInt(int i);
extern void printNewlines(int l);

export int main() {
    int x = 0;
    int y = 10;
    int z;
    do {
        x = x + 1;
        y = y - 1;
        z = x;
        do {
            printInt(z);
            z = z + 1;
        } while (z < y);
        printNewlines(1);
        printInt(x);
        printInt(y);
        printNewlines(1);
    } while (x <= y);
    return 0;
}
