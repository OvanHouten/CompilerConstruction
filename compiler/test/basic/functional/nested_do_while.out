12345678
19
234567
28
3456
37
45
46
5
55
6
64